src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -650,50 +650,77 @@ public static string ConvertPathToDocumentUri(string path)
+                // original path for colon since a char search (no string culture involved) is
+                // faster than a string search.  If found, then lowercase the associated drive letter.
                 if (path.Contains(':'))
                 {
-                    // Start at index 1 to avoid an index out of range check when accessing index - 1.
-                    // Also, if the colon is at index 0 there is no drive letter before it to lower case.
-                    for (int i = 1; i < docUriStrBld.Length - 2; i++)
+                    // A valid, drive-letter based path converted to URI form needs to be prefixed
+                    // with a / to indicate the path is an absolute path.
+                    docUriStrBld.Append("/");
+                    int prefixLen = docUriStrBld.Length;
+
+                    docUriStrBld.Append(escapedPath);
+
+                    // Uri.EscapeDataString goes a bit far, encoding \ chars. Also, VSCode wants / instead of \.
+                    docUriStrBld.Replace("%5C", "/");
+
+                    // Find the first colon after the "file:///" prefix, skipping the first char after
+                    // the prefix since a Windows path cannot start with a colon. End the check at
+                    // less than docUriStrBld.Length - 2 since we need to look-ahead two characters.
+                    for (int i = prefixLen + 1; i < docUriStrBld.Length - 2; i++)
                     {
                         if ((docUriStrBld[i] == '%') && (docUriStrBld[i + 1] == '3') && (docUriStrBld[i + 2] == 'A'))
                         {
                             int driveLetterIndex = i - 1;
                             char driveLetter = char.ToLowerInvariant(docUriStrBld[driveLetterIndex]);
-                            docUriStrBld.Replace(path[driveLetterIndex], driveLetter, driveLetterIndex, 1);
+                            docUriStrBld.Replace(docUriStrBld[driveLetterIndex], driveLetter, driveLetterIndex, 1);
                             break;
                         }
                     }
                 }
+                else
+                {
+                    // This is a Windows path without a drive specifier, must be either a relative or UNC path.
+                    int prefixLen = docUriStrBld.Length;
+
+                    docUriStrBld.Append(escapedPath);
+
+                    // Uri.EscapeDataString goes a bit far, encoding \ chars. Also, VSCode wants / instead of \.
+                    docUriStrBld.Replace("%5C", "/");
 
-                // Uri.EscapeDataString goes a bit far, encoding \ chars. Also, VSCode wants / instead of \.
-                docUriStrBld.Replace("%5C", "/");
+                    // The proper URI form for a UNC path is file://server/share.  In the case of a UNC
+                    // path, remove the path's leading // because the file:// prefix already provides it.
+                    if ((docUriStrBld.Length > prefixLen + 1) &&
```

Comment:
Instead of removing, why not just use the `if (path.Contains(':'))` conditional to add either 2 slashes or 3?

Reply:
This is because of relative paths.

Reply:
I could do that. I was just trying to lay down the parts of the URI scheme in an order that made sense - at least to me :-).  And I wasn't as worried about replaces that don't re-allocate memory.